packages/timeline-state-resolver/src/integrations/tricaster/triCasterShortcutStateConverter.ts

```diff
@@ -0,0 +1,191 @@
+import {
+	TriCasterAudioChannelName,
+	TriCasterDelegateName,
+	TriCasterInputName,
+	TriCasterKeyerName,
+	TriCasterLayerName,
+	TriCasterMixEffectName,
+	TriCasterMixOutputName,
+} from 'timeline-state-resolver-types'
+import { ElementCompact, xml2js } from 'xml-js'
+import {
+	fillRecord,
+	TriCasterMixEffectState,
+	TriCasterAudioChannelState,
+	TriCasterInputState,
+	TriCasterKeyerState,
+	TriCasterLayerState,
+	TriCasterMixOutputState,
+	TriCasterState,
+} from './triCasterStateDiffer'
+import { CommandName, TriCasterGenericCommandName } from './triCasterCommands'
+
+type ShortcutStates = {
+	[key: string]: string | number
+}
+
+export class TriCasterShortcutStateConverter {
+	constructor(
+		private readonly meNames: TriCasterMixEffectName[],
+		private readonly inputNames: TriCasterInputName[],
+		private readonly audioChannelNames: TriCasterAudioChannelName[],
+		private readonly layerNames: TriCasterLayerName[],
+		private readonly keyerNames: TriCasterKeyerName[],
+		private readonly mixOutputNames: TriCasterMixOutputName[]
+	) {}
+
+	getTriCasterStateFromShortcutState(shortcutStatesXml: string): TriCasterState {
+		const parsedStates = xml2js(shortcutStatesXml, { compact: true }) as ElementCompact
+		const shortcutStateElement = parsedStates.shortcut_states?.shortcut_state as ElementCompact | ElementCompact[]
+		const shortcutStates = this.extractShortcutStates(shortcutStateElement)
+
+		return {
+			mixEffects: this.parseMixEffectsState(shortcutStates),
+			inputs: this.parseInputsState(),
+			outputs: this.parseMixOutputsState(shortcutStates),
+			audioChannels: this.parseAudioChannelsState(shortcutStates),
+			isRecording: this.parseNumber(shortcutStates, [], CommandName.RECORD_TOGGLE) === 1,
+			isStreaming: this.parseNumber(shortcutStates, [], CommandName.STREAMING_TOGGLE) === 1,
+		}
+	}
+
+	private extractShortcutStates(shortcutState: ElementCompact | ElementCompact[]): ShortcutStates {
```

Comment:
Why not just have the parameter as `shorctutStates: ElemetCompact[]` and then instead of checking if it's an array or not, we just check if there is exactly one entry? (Which I am not entirely sure would even be needed - if it's not then you can get rid of some duplication in this method)